File: frontend/src/api/clue.ts
import { http } from './http'

export interface Clue {
  id?: number
  title: string
  content: string
  location: string
  contact: string
  image: string
  createTime?: string
}

export async function submitClue(data: Clue) {
  const res = await http.post('/api/clue', data)
  return res.data as { code: number; message: string; data: boolean }
}

export async function getClueList() {
  const res = await http.get('/api/clue')
  return res.data as { code: number; message: string; data: Clue[] }
}

File: backend/src/main/java/com/xxx/animal/entity/Clue.java
package com.xxx.animal.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("clue")
public class Clue {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String content;
    private String location;
    private String contact;
    private String image;
    private Integer status; // 0: Pending, 1: Processed
    private Long userId;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
